clang-tools-extra/test/clang-tidy/checkers/bsl-for-loop.cpp

```diff
@@ -0,0 +1,55 @@
+// RUN: %check_clang_tidy %s bsl-for-loop %t
+
+// A6-5-2
+void foo()
+{
+	int y = 0;
+
+	long double l = 0;
+	// CHECK-MESSAGES: :[[@LINE-1]]:2: warning: float type not allowed (variable declaration) [bsl-for-loop]
+
+	double d = 0;
+	// CHECK-MESSAGES: :[[@LINE-1]]:2: warning: float type not allowed (variable declaration) [bsl-for-loop]
+
+	double d2 = 0.0;
+	// CHECK-MESSAGES: :[[@LINE-1]]:2: warning: float type not allowed (variable declaration) [bsl-for-loop]
+
+	double d3;
+	// CHECK-MESSAGES: :[[@LINE-1]]:2: warning: float type not allowed (variable declaration) [bsl-for-loop]
+
+	d3 = 0;
+
+	int f = 0.0;
+	// CHECK-MESSAGES: :[[@LINE-1]]:10: warning: float type not allowed (literal) [bsl-for-loop]
+
+	for (int i = 0; i < 3; ++i) {}
+
+	for (int x = 0; x < 3 && y < 15; x++, y++) {}
```

Comment:
I would add more variations on this. Specifically, the && is not an issue... its the use of "," in the for loop that is an issue. You can use "," in both the initialization step and the increment step. I would make sure that && without "," is good, and that "," without the && in both the init and inc are both detected

Reply:
Do you mean you can't use "," in both the initialization step and increment step? So for a single loop counter, if you initialized two counters but actually only incremented one, that would still be a violation 

Also, should I split this check since we changed half to make it "no floating point," not just no floating point in for loops? Did you have any thoughts on renaming the check?